[BUG] IplImage references in imagealgobuf.h break in the absence of OpenCV

**Describe the bug**
Background information is available at [Notcurses #541](https://github.com/dankamongmen/notcurses/issues/541). I was writing some code against OIIO 
as shipped in Debian Unstable's 2.1.13.0~dfsg0, and Fedora 32's 2.1.13.0-2.fc32. I had the same result on both (note that neither is built against OpenCV). My code `#include`d the following:

```c
#include <OpenImageIO/filter.h>
#include <OpenImageIO/version.h>
#include <OpenImageIO/imageio.h>
#include <OpenImageIO/imagebuf.h>
#include <OpenImageIO/imagebufalgo.h>
```

and linked `-lOpenImageIO`, following the various suggestions of `pkg-config`. No matter what I did, I got the following linker failure:

```
/usr/bin/ld: libnotcurses.so.1.3.2: undefined reference to `OpenImageIO_v2_1::ImageBufAlgo::from_IplImage(IplImage const*, OpenImageIO_v2_1::TypeDesc)'
collect2: error: ld returned 1 exit status
```

I had no direct calls to this, but there are `inline` functions in `imagealgobuf.h` which reference them. Here's the problem: at the top of this file, you have

```c
#if !defined(__OPENCV_CORE_TYPES_H__) && !defined(OPENCV_CORE_TYPES_H)
struct IplImage;  // Forward declaration; used by Intel Image lib & OpenCV
```

and later you have

```c
ImageBuf OIIO_API from_IplImage (const IplImage *ipl,
inline bool from_IplImage (ImageBuf &dst, const IplImage *ipl,
OIIO_API IplImage* to_IplImage (const ImageBuf &src);
```

If OpenCV headers (in particular, `<opencv2/core/types_c.h>`) are `#include`d, this compiles up just fine. In their absence, unadorned `IplImage` is a distinct type from `struct IplImage`. They mangle to:

`_ZN16OpenImageIO_v2_112ImageBufAlgo13from_IplImageEPK8IplImageNS_8TypeDescE`

and

`_ZN16OpenImageIO_v2_112ImageBufAlgo13from_IplImageEPK9_IplImageNS_8TypeDescE`

Note `EPK8` v `EPK9`. Hence the missing symbol; hence my failed link.

I verified that changing these to `struct IplImage*`, rebuilding OIIO, reinstalling OIIO, and rebuilding my program worked. The problem with that, of course, is that if OpenCV **is** present, you'll draw a "referenced typedef'd type with `struct`" complaint from the compiler.

I submit the following (I have it in a branch, ready to PR):

```c
#if !defined(__OPENCV_CORE_TYPES_H__) && !defined(OPENCV_CORE_TYPES_H)
struct IplImage;
#define IplImage struct IplImage
```


**To Reproduce**
Steps to reproduce the behavior:
1. Install OpenImageIO-devel on Fedora 32, or libopenimageio-dev on Debian Unstable
2. Ensure OpenCV is **not** installed
3. Write code using IBA
4. Try and fail to link it

**Expected behavior**
My program links.

**Evidence**
See above. I'm happy to provide more information.

**Platform information:**
 - OIIO branch/version: 2.1.13
 - OS: Fedora 32, Debian Unstable
 - C++ compiler: g++ 9.3.0
 - Any non-default build flags when you build OIIO:

https://github.com/dankamongmen/oiio/tree/dankamongmen/opencv-struct-link

--- a/.github/workflows/workflow.yml
+++ b/.github/workflows/workflow.yml
@@ -109,7 +109,7 @@ jobs:
 
   linux-oldest:
     # Oldest versions of the dependencies that we can muster, and various
-    # things disabled (no SSE, no OCIO available, don't embed plugins)
+    # things disabled (no SSE, OCIO, or OpenCV, don't embed plugins)
     name: "Linux oldest/hobbled: gcc4.8/C++11 py2.7 boost-1.66 exr-2.2 no-sse no-ocio"
     runs-on: ubuntu-latest
     container:
@@ -125,6 +125,7 @@ jobs:
           USE_SIMD: 0
           USE_JPEGTURBO: 0
           USE_OPENCOLORIO: 0
+          USE_OPENCV: 0
           EMBEDPLUGINS: 0
           OPENEXR_BRANCH: v2.2.0
         run: |

--- a/src/include/OpenImageIO/imagebufalgo.h
+++ b/src/include/OpenImageIO/imagebufalgo.h
@@ -2053,6 +2053,12 @@ inline bool capture_image (ImageBuf &dst, int cameranum = 0,
     return !dst.has_error();
 }
 
+
+
+#if defined(__OPENCV_CORE_TYPES_H__) || defined(OPENCV_CORE_TYPES_H)
+// These declarations are only visible if the OpenCV headers have already
+// been encountered.
+
 // DEPRECATED(2.0). The OpenCV 1.x era IplImage-based functions should be
 // avoided, giving preference to from_OpenCV.
 ImageBuf OIIO_API from_IplImage (const IplImage *ipl,
@@ -2068,6 +2074,7 @@ inline bool from_IplImage (ImageBuf &dst, const IplImage *ipl,
 // avoided, giving preference to from_OpenCV.
 OIIO_API IplImage* to_IplImage (const ImageBuf &src);
 
+#endif
 
 
 

--- a/src/libOpenImageIO/imagebufalgo_opencv.cpp
+++ b/src/libOpenImageIO/imagebufalgo_opencv.cpp
@@ -46,10 +46,11 @@
 OIIO_NAMESPACE_BEGIN
 
 
+namespace ImageBufAlgo {
 
 // Note: DEPRECATED(2.0)
 ImageBuf
-ImageBufAlgo::from_IplImage(const IplImage* ipl, TypeDesc convert)
+from_IplImage(const IplImage* ipl, TypeDesc convert)
 {
     pvt::LoggedTimer logtime("IBA::from_IplImage");
     ImageBuf dst;
@@ -123,7 +124,7 @@ ImageBufAlgo::from_IplImage(const IplImage* ipl, TypeDesc convert)
 
 // Note: DEPRECATED(2.0)
 IplImage*
-ImageBufAlgo::to_IplImage(const ImageBuf& src)
+to_IplImage(const ImageBuf& src)
 {
     pvt::LoggedTimer logtime("IBA::to_IplImage");
 #ifdef USE_OPENCV
@@ -219,6 +220,7 @@ RBswap(ImageBuf& R, ROI roi, int nthreads)
     return true;
 }
 
+}  // end namespace ImageBufAlgo
 
 
 ImageBuf

